### PR TITLE
edit allowance based on new collaborator props

### LINKED ISSUE #1519: Collaborators descriptions role allowance
BE:
- [x] Update CollaboratorProps to be sections: 'all' | 'none' | Record<string ,{tableData:boolean,descriptions:boolean}>
- [x] in current data migration, update current sections to {tableData:true,descriptions:true}
- [x] Update current auth middleware to authorize collaborators to edit table data and descriptions (treat them differently if they'are not now)

FE:
- [x] User edit ui:  update current sections edit to reflect the possibility to edit tableData and descriptions
- [x] Descriptions edit: update edit allowance based on new collaborator props
- [x] Table data edit: update edit allowance based on new collaborator props

OJ: Add functionality to allow contributors to edit text sections but not change data in the reporting tables (Örjan) (TODO: REFINE access level)

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1519

### CODE REVIEW

File: src/client/pages/AssessmentSection/AssessmentSection.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { AssessmentName, AssessmentNames } from '@meta/assessment'
 import { useAssessmentSection } from '@client/store/assessment'
 import { useTableSections } from '@client/store/pages/assessmentSection'
 import { useIsSectionDataEmpty } from '@client/store/pages/assessmentSection/hooks'
-import { useCanEditSection } from '@client/store/user'
+import { useCanEditSectionDescriptions, useCanEditSectionTableData } from '@client/store/user/hooks'
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/AssessmentSection.tsx]: let's shorten the names. useCanEditDescriptions and useCanEditTableData

File: src/client/pages/AssessmentSection/AssessmentSection.tsx [hunk 2]
```diff
@@ -50,7 +53,9 @@ const AssessmentSection: React.FC<Props> = (props: Props) => {
 
       <SectionHeader assessmentName={assessmentName} sectionName={sectionName} disabled={disabled} />
 
-      {!panEuropean && <Descriptions sectionName={sectionName} descriptions={descriptions} disabled={disabled} />}
+      {!panEuropean && (
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/AssessmentSection.tsx]: also, let's remove this panEuropean check. we will allow editing panEuropean data

File: src/client/store/user/hooks/index.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ export const useUserCountries = (): Array<CountryIso> => {
   return user?.roles.map((role) => role.countryIso)
 }
 
-export const useCanEditSection = (sectionName?: string) => {
+export const useCanEditSection = (sectionName?: string, permission?: CollaboratorEditPropertyType) => {
```
Comment (minotogna) [inline src/client/store/user/hooks/index.ts]: this should not be exported anymore

**Review by minotogna — CHANGES_REQUESTED** [review]
see my previous comments

**Review by sorja — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**